src/EventStore.Core.Tests/ClientAPI/ExpectedVersion64Bit/subscribe_to_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue.cs

```diff
@@ -0,0 +1,58 @@
+﻿using EventStore.ClientAPI;
+using EventStore.ClientAPI.Common.Utils;
+using EventStore.Core.Services;
+using NUnit.Framework;
+using System;
+using System.Threading;
+using ResolvedEvent = EventStore.ClientAPI.ResolvedEvent;
+
+namespace EventStore.Core.Tests.ClientAPI.ExpectedVersion64Bit
+{
+    [TestFixture]
+    [Category("ClientAPI"), Category("LongRunning")]
+    public class subscribe_to_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue : MiniNodeWithExistingRecords
+    {
+        private const string StreamName = "subscribe_to_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue";
+        private const long intMaxValue = (long)int.MaxValue;
+
+        private string _linkedStreamName = "linked-" + StreamName;
+        private Guid _event1Id;
+
+        private readonly AutoResetEvent _resetEvent = new AutoResetEvent(false);
+        private ResolvedEvent _receivedEvent;
+
+        public override void WriteTestScenario()
+        {
+            var event1 = WriteSingleEvent(StreamName, intMaxValue + 1, new string('.', 3000));
+            WriteSingleEvent(StreamName, intMaxValue + 2, new string('.', 3000));
+            _event1Id = event1.EventId;
+        }
+
+        public override void Given()
+        {
+            _store = BuildConnection(Node);
+            _store.ConnectAsync().Wait();
+
+            _store.SubscribeToStreamAsync(_linkedStreamName, true, HandleEvent).Wait();
+            _store.AppendToStreamAsync(_linkedStreamName, -1,
```

Comment:
We probably want to use `ExpectedVersion.NoStream` here instead of `-1` for consistency.